src/AppNavigations/AppNavigations.js

```diff
@@ -1,15 +1,23 @@
 import { Routes, Route } from "react-router-dom";
 import Homepage from "../Pages/HomePage/Homepage";
 import DisplayComponents from "../Components/DisplayComponents/DisplayComponents";
+import EditWarehouse from "../Components/EditWarehouse/EditWarehouse";
 
 const AppNavigations = () => {
+  const handleClick = () => {
+    alert("Button clicked!");
+  };
+
```

Comment:
The handleClick function should be moved to the EditWarehouse.js file because you're passing it to the form's cancel button within that component. There's no need to pass it as props from AppNavigations.js. Keeping the function within EditWarehouse.js will make the component more self-contained and improve maintainability.

In AppNavigations.js, you can remove the handleClick function and pass it directly within EditWarehouse.js to the cancel button, which will simplify your routing logic and reduce unnecessary prop drilling.

Reply:
Makes sense. I have removed it completely and defined a new handleCancel function:

```
 const handleCancel = () => {
    navigate(-1);
  };
```

Reply:
this takes you one step back in the navigation stack